test/e2e/singlecluster/statefulset_test.go

```diff
@@ -126,12 +126,12 @@ var _ = ginkgo.Describe("Stateful set integration", func() {
 				conflictingWorkload := &kueue.Workload{}
 				gomega.Expect(k8sClient.Get(ctx, conflictingWlLookupKey, conflictingWorkload)).To(gomega.Succeed())
 				gomega.Expect(createdWorkload.Name).ToNot(gomega.Equal(conflictingWorkload.Name))
-				util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, conflictingStatefulSet, true, util.LongTimeout)
```

Comment:
Please provide some explanation why it flaked, and why swapping the statements will fix the issue.

Reply:
The StatefulSet is being removed immediately. However, the workload is waiting for the pods to be removed because the workload's ownerReferences is a list of pods.

Reply:
Ah, I see, and previously for Workload we only used default Timeout=5s. Makes sense.